RecoTracker/CkfPattern/python/CkfTrackCandidatesBHM_cff.py

```diff
@@ -2,30 +2,31 @@
 
 #special propagator
 from TrackingTools.GeomPropagators.BeamHaloPropagator_cff import *
-
 from RecoTracker.CkfPattern.CkfTrajectoryBuilder_cff import *
 import  TrackingTools.TrajectoryFiltering.TrajectoryFilter_cff
-ckfTrajectoryFilterBeamHaloMuon = TrackingTools.TrajectoryFiltering.TrajectoryFilter_cff.CkfBaseTrajectoryFilter_block.clone()
-import copy
 from RecoTracker.CkfPattern.CkfTrajectoryBuilder_cfi import *
-# clone the trajectory builder
-CkfTrajectoryBuilderBeamHalo = copy.deepcopy(CkfTrajectoryBuilder)
-import copy
 from RecoTracker.CkfPattern.CkfTrackCandidates_cfi import *
-# generate CTF track candidates ############
-beamhaloTrackCandidates = copy.deepcopy(ckfTrackCandidates)
 
-ckfTrajectoryFilterBeamHaloMuon.minimumNumberOfHits = 4
-ckfTrajectoryFilterBeamHaloMuon.minPt = 0.1
-ckfTrajectoryFilterBeamHaloMuon.maxLostHits = 3
-ckfTrajectoryFilterBeamHaloMuon.maxConsecLostHits = 2
+ckfTrajectoryFilterBeamHaloMuon = TrackingTools.TrajectoryFiltering.TrajectoryFilter_cff.CkfBaseTrajectoryFilter_block.clone(
+    minimumNumberOfHits = 4,
+    minPt               = 0.1,
+    maxLostHits         = 3,
+    maxConsecLostHits   = 2
+)
 
-CkfTrajectoryBuilderBeamHalo.propagatorAlong = 'BeamHaloPropagatorAlong'
-CkfTrajectoryBuilderBeamHalo.propagatorOpposite = 'BeamHaloPropagatorOpposite'
-CkfTrajectoryBuilderBeamHalo.trajectoryFilter.refToPSet_ = 'ckfTrajectoryFilterBeamHaloMuon'
-beamhaloTrackCandidates.src = cms.InputTag('beamhaloTrackerSeeds')
-beamhaloTrackCandidates.NavigationSchool = 'BeamHaloNavigationSchool'
-beamhaloTrackCandidates.TransientInitialStateEstimatorParameters.propagatorAlongTISE = 'BeamHaloPropagatorAlong'
-beamhaloTrackCandidates.TransientInitialStateEstimatorParameters.propagatorOppositeTISE = 'BeamHaloPropagatorOpposite'
-beamhaloTrackCandidates.TrajectoryBuilderPSet.refToPSet_ = 'CkfTrajectoryBuilderBeamHalo'
+# clone the trajectory builder
+CkfTrajectoryBuilderBeamHalo = CkfTrajectoryBuilder.clone(
```

Comment:
CkfTrajectoryBuilderBeamHalo was a ```deepcopy``` of CkfTrajectoryBuilder, and here it becomes a ```clone``` of it. 

Honestly, I'm not expert in python enough to fully understand the difference between them. In the original python code both ```deepcopy``` and ```clone``` were used.

Also for my own understanding, I'd ask @davidlange6 and @slava77 whether they know about any possible difference between them, and if they think that using ```clone``` instead of ```deepcopy``` is equivalent, or justified here.

(Same for beamhaloTrackCandidates below)

Reply:
> CkfTrajectoryBuilderBeamHalo was a `deepcopy` of CkfTrajectoryBuilder, and here it becomes a `clone` of it.
> 
> Honestly, I'm not expert in python enough to fully understand the difference between them. In the original python code both `deepcopy` and `clone` were used.
> 
> Also for my own understanding, I'd ask @davidlange6 and @slava77 whether they know about any possible difference between them, and if they think that using `clone` instead of `deepcopy` is equivalent, or justified here.
> 
> (Same for beamhaloTrackCandidates below)

@makortel @Dr15Jones 
please advise if there is a substantive difference

Reply:
I believe we prefer `clone` over `deepcopy`.

Reply:
> I believe we prefer `clone` over `deepcopy`.

Great! Thank you @Dr15Jones